### PR TITLE
syntax_tools: Fix pos during reverting

### PR DESCRIPTION
**Pull request by maxno-kivra**
This fixes/improves the position (really an `erl_anno` annotation) handing in `erl_syntax`.

I discovered today that reverting any `#wrapper` will throw away any modifications to the `pos`, which is fixed in two ways here. First off, make sure `set_pos` also update the inner `tree`. Secondly by always using `revert_root`.
I think either change will fix this, but I prefer having both, as it is nice to keep the `pos` in sync at all times.

While debugging this (by parsing `epp` with `epp` and comparing that with `erl_syntax:revert(erl_syntax_lib:annotate_bindings(...))`), I noticed that the list reverting was also throwing away `pos`/annotations. So I fixed that, mostly anyway. The remaining issues are hard to get around, but the changes reduces the diff.

### CODE REVIEW

File: lib/syntax_tools/src/erl_syntax.erl [hunk 1]
```diff
@@ -930,8 +930,12 @@ set_pos(Node, Pos) ->
...
 	#tree{attr = Attr} ->
 	    Node#tree{attr = Attr#attr{pos = Pos}};
-	#wrapper{attr = Attr} ->
-	    Node#wrapper{attr = Attr#attr{pos = Pos}};
+	#wrapper{attr = Attr, tree = {error, {_, Module, Reason}}} ->
+	    Node#wrapper{attr = Attr#attr{pos = Pos}, tree = {error, {Pos, Module, Reason}}};
+    #wrapper{attr = Attr, tree = {warning, {_, Module, Reason}}} ->
+        Node#wrapper{attr = Attr#attr{pos = Pos}, tree = {warning, {Pos, Module, Reason}}};
+	#wrapper{attr = Attr, tree = Tree} ->
+	    Node#wrapper{attr = Attr#attr{pos = Pos}, tree = setelement(2, Tree, Pos)};
```
Comment (jhogberg) [inline lib/syntax_tools/src/erl_syntax.erl:938]: Nitpick: the indentation is a bit off. While you're at it, you can convert tabs to 8 spaces here.

File: lib/syntax_tools/src/erl_syntax.erl [hunk 2]
```diff
@@ -2828,7 +2838,7 @@ compact_list(Node) ->
 					  copy_attrs(Node,
 						     Node1));
 			_ ->
-			    Node
```
Comment (jhogberg) [inline lib/syntax_tools/src/erl_syntax.erl:2831]: Nitpick: unrelated whitespace change. Please remove the others like it as well.

**Review by jhogberg — CHANGES_REQUESTED** [review]
Thanks for the PR, and sorry for the delay in getting around to this. It looks good aside from a few nits, and I'd like it to be rebased to `maint` for inclusion in OTP 26.1. :)

File: lib/syntax_tools/src/erl_syntax.erl (hunk 2 above)
Comment (maxno-kivra) [reply lib/syntax_tools/src/erl_syntax.erl:2831]: Do you mean all the random trailing spaces? They were removed by my editor, and I suspect that will be the case for most editors. So in order to reduce the next diff, I would like to keep the commit.

File: lib/syntax_tools/src/erl_syntax.erl (hunk 1 above)
Comment (maxno-kivra) [reply lib/syntax_tools/src/erl_syntax.erl]: I think I got rid of all the tabs now.

**Comment by maxno-kivra** [comment]
Now it's my turn to apologise for the delay 😉. I'm back from vacation and have rebased on latest master and replied to your comments.

File: lib/syntax_tools/src/erl_syntax.erl (hunk 2 above)
Comment (frej) [reply lib/syntax_tools/src/erl_syntax.erl:2831]: Changes which are not related to what the patch claims to do, mask the real reason for a code line looking like it does. If I'm trying to find a bug, I want git blame to point to the last functional change, not a completely unrelated patch doing a cosmetic change. I guess that's why the [contributor's guide](https://github.com/erlang/otp/blob/master/CONTRIBUTING.md) asks you to not fix preexisting whitespace errors in otherwise untouched source lines.

Comment (maxno-kivra) [reply lib/syntax_tools/src/erl_syntax.erl]: I removed the commit that removes the trailing whitespace, but for posterity I'd like to mention the `--ignore-rev`/`--ignore-revs-file` flags to `git blame` which allows you to ignore/automatically re-blame.

**Comment by maxno-kivra** [comment]
@jhogberg I removed the white space commit a while ago, but I guess it got lost in the ether. 

Anyway, could you please take another look at this? 🙏

**Comment by maxno-kivra** [comment]
Ping @jhogberg please take another look

**Comment by maxno-kivra** [comment]
@jhogberg @frej either close this or re-review it now that I have removed the white space commit. It fixes a bug and adds a test to prove it.

I understand you have limited bandwidth, but please at least communicate that you don't have time. Silence for weeks just makes one wonder if you only have one shot to get you PR approved/merged. If you fail the first review you'll just get ignored. I really hope I'm wrong about that, so please prove me wrong here. Thanks 🙏

**Comment by jhogberg** [comment]
> Silence for weeks just makes one wonder if you only have one shot to get you PR approved/merged.

Sorry, it was my turn to go on vacation just prior to you getting back. I returned to work a bit after you pinged me and missed the notification. Oops. :smile:

I've added it to our nightly runs, and will merge it tomorrow if nothing explodes.

**Comment by jhogberg** [comment]
Merged, thanks again for the PR! :)

**Comment by maxno-kivra** [comment]
❤️
